jeta/base.py

```diff
@@ -0,0 +1,65 @@
+from abc import ABC
+
+from jax import random
+
+
+class BaseLearner(ABC):
+    """Base Learner class
+
+    Attributes:
+        batch_size(int): batch size for training the algorithm on
+        alpha(float): Learning rate
+        fas(int): Fast adaptation steps
+        seed(int): integer value for generating key for random generators
+        first_order(bool): True/False, for extending MAML to FO-MAML
+
+    """
+
+    def __init__(
+        self, batch_size: int = 32, alpha: float = 0.001, fas: int = 1, **kwargs
+    ):
+
+        self.batch_size = batch_size
+        self.alpha = alpha
+        self.fas = fas
+        self.losses = []
+        self.seed = kwargs["seed"] if "seed" in kwargs else None
+        self.first_order = kwargs["first_order"] if "first_order" in kwargs else None
+        if self.seed is not None:
+            self.key = random.PRNGKey(self.seed)
+
+    def train(self, params, batch):
+        """train algorithm
+
+        Args:
+            params(ndarray): initial parameters for the model
```

Comment:
I am not sure if the type for params will be ndarray.

Reply:
I think the type of the params should be a FrozenDict and not ndarray.  or am I missing anything? 

cc: @veds12

Reply:
I agree @abhi-glitchhg. 

@akagam1 please fix this